### PR TITLE
add an example for `careless.stats` for the dw manuscript revisions

### PR DESCRIPTION
**Pull request by hkwang**

### CODE REVIEW

**Comment by hkwang** [comment]
Context: To respond to a reviewer comment about adding more machine learning metrics, Doeke asked me to create an example notebook explaining `careless.stats`. 

I'm unable to add reviewers. This would be primarily @DHekstra for content and @kmdalton for formatting: Kevin, this is notably the only `.ipynb` in here; let me know if there is a better place for this

**Comment by hkwang** [comment]
In response to @DHekstra comments in Slack, I've added I/sigI, completeness, Rsplit, image_CC. I am not used to the last two metrics, so I invite @DHekstra to provide commentary for them. I will also submit a PR to careless.stats to allow users to set the figure legend position.

**Comment by kmdalton** [comment]
If I understand correctly, this PR adds an IPython Notebook file which contains examples of running command line applications. I think only some simple diagnostic plots of the loss trajectory are coded in python, and these could equally be generated using `careless.plot_history`. I don't want to get too philosophical here, but in my opinion `.ipynb` is the wrong tool for command line tutorials. It adds a whole bunch of overhead on the user's end and doesn't always render cleanly in the browser. Along these lines, wouldn't this be better as a short markdown file? It would be easier to maintain as well.

**Comment by hkwang** [comment]
Sure, I'll push a .md file.

**Comment by hkwang** [comment]
I converted the `.ipynb` to a `.md` and pushed changes in the above commit. I also edited some plots following https://github.com/rs-station/careless/pull/187/commits/19646ea07a86d3ad19cc6efd6bb7ab6b3e7ccf83.

**Comment by hkwang** [comment]
the above commit involved removing lots of extraneous files related to refinement of the merged data. There are, however, more extraneous files we can consider removing. @kmdalton , what are: 

```
careless_stats/thermolysin_dw/thl_1p8A_grid_scale.data-00000-of-00001
careless_stats/thermolysin_dw/thl_1p8A_grid_scale.index
careless_stats/thermolysin_dw/thl_1p8A_grid_structure_factor.data-00000-of-00001
careless_stats/thermolysin_dw/thl_1p8A_grid_structure_factor.index
```

**Comment by kmdalton** [comment]
> the above commit involved removing lots of extraneous files related to refinement of the merged data. There are, however, more extraneous files we can consider removing. @kmdalton , what are:
> 
> ```
> careless_stats/thermolysin_dw/thl_1p8A_grid_scale.data-00000-of-00001
> careless_stats/thermolysin_dw/thl_1p8A_grid_scale.index
> careless_stats/thermolysin_dw/thl_1p8A_grid_structure_factor.data-00000-of-00001
> careless_stats/thermolysin_dw/thl_1p8A_grid_structure_factor.index
> ```

Let's remove any and all extraneous files from this PR. I think all that should be needed are 
1) any input mtz files for stats commands
2) any images we wish to display in the markdown file

Any CSV files generated by careless stats don't need to be saved. They can just be inlined if necessary. Logs don't need to be included.

File: careless_stats/careless_stats.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+3. [Machine learning metrics](#Machinelearningmetrics)
+
+
+## Elementary crystallographic statistics <a name="Elementarycrystallographicstatistics"></a>
+
+### Completeness and multiplicity
+`careless.stats` also computes standard crystallography statistics, such as completeness and $I/\sigma(I)$. 
+
+```
+>careless.completeness ./thermolysin_dw/thl_1p8A_grid_both.mtz -o ./thermolysin_dw/completeness.csv -b 10
```
Comment (kmdalton) [inline careless_stats/careless_stats.md]: angle brackets should not be used to indicate the command promp, because they preclude copying and pasting the command.

File: careless_stats/careless_stats.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
+# Using `careless.stats`
+
+This notebook describes all the machine learning and data crossvalidation metrics that `careless` can compute. As an example, we describe the metrics from the analysis of anomalous signal from a thermolysin serial femtosecond crystallography dataset, as done in [Hekstra, Wang et al. 2025](https://www.biorxiv.org/content/10.1101/2024.07.22.604476v1). This notebook heavily draws on explanations from [Aldama, Dalton, and Hekstra 2023](https://journals.iucr.org/d/issues/2023/09/00/qi5002/index.html), which is good prerequisite reading for this notebook; this notebook also breaks down the common validation scripts found in `dw-examples`. 
+
+# Table of Contents
+1. [Elementary crystallographic statistics](#Elementarycrystallographicstatistics)
```
Comment (kmdalton) [inline careless_stats/careless_stats.md]: these hyperlinks links don't work properly in my browser.

File: careless_stats/careless_stats.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+`careless.stats` also computes standard crystallography statistics, such as completeness and $I/\sigma(I)$. 
+
+```
+>careless.completeness ./thermolysin_dw/thl_1p8A_grid_both.mtz -o ./thermolysin_dw/completeness.csv -b 10
+```
+![image](./thermolysin_dw/completeness.png)
+
+Now, in python: 
+```
+pd.read_csv("./thermolysin_dw/completeness.csv")
```
Comment (kmdalton) [inline careless_stats/careless_stats.md]: this doesn't need to be done in python, the CLI can make a graphic for you.

File: careless_stats/careless_stats.md [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+```
+![image](./thermolysin_dw/pd_completeness.png)
+
+The completeness of the last resolution bin is 0.88, and the overall completeness is 0.984.
```
Comment (kmdalton) [inline careless_stats/careless_stats.md]: could we add the CLI output here which should print the table of completeness values over bins?

File: careless_stats/careless_stats.md [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+
+![image](./thermolysin_dw/imagecc_careless.png)
+
+The above plot displays the CCpred for each image. Images with low CCpred are either misindexed or their scales are poorly modeled. 
+
+
+## Machine learning metrics <a name="Machinelearningmetrics"></a>
+
+`careless` is a package for scaling and merging crystallographic data using variational inference. Every time I process a dataset using `careless`, I will learn the distribution of the structure factor amplitudes and the scales for each miller index, given the intensity of each observed reflection in the input dataset. The scale function is implemented as a neural network, whose parameters are optimized by minimizing an objective function ('loss'). 
+
```
Comment (kmdalton) [inline careless_stats/careless_stats.md]: Replace with `careless.plot_history`

File: careless_stats/careless_stats.md [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+training_metrics.plot(x="step",y="NLL", ax=ax,loglog=True)
+training_metrics.plot(x="step",y="NLL_val", ax=ax, loglog=True)
+training_metrics.plot(x="step",y="F KLDiv", ax=ax, loglog=True)
+plt.savefig("./thermolysin_dw/loss_components.png")
+plt.show()
+```
+![image](./thermolysin_dw/loss.png)
+
+
+We see that the loss goes down over the 30000 steps of training. This loss is smooth. If the loss is jagged, then I advise increasing `--mc-samples` for smoother training. Training takes about 17.5 minutes as according to the below log:
```
Comment (kmdalton) [inline careless_stats/careless_stats.md]: remove runtime info (hardware dependent)

File: careless_stats/careless_stats.md [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+
+```
```
Comment (kmdalton) [inline careless_stats/careless_stats.md]: cut this block

File: careless_stats/thermolysin_dw/ccanom_overall.csv [hunk 8]
```diff
@@ -0,0 +1,4 @@
+,file,repeat,Resolution Range (Å),bin,Spacegroup,CCanom
```
Comment (kmdalton) [inline careless_stats/thermolysin_dw/ccanom_overall.csv]: delete file

File: careless_stats/thermolysin_dw/ccanom_overall.png
Comment (kmdalton) [inline careless_stats/thermolysin_dw/ccanom_overall.png:1]: delete

File: careless_stats/thermolysin_dw/ccanom_res.csv
Comment (kmdalton) [inline careless_stats/thermolysin_dw/ccanom_res.csv:1]: delete

File: careless_stats/thermolysin_dw/cchalf_overall.csv
Comment (kmdalton) [inline careless_stats/thermolysin_dw/cchalf_overall.csv:1]: delete

File: careless_stats/thermolysin_dw/cchalf_res.csv
Comment (kmdalton) [inline careless_stats/thermolysin_dw/cchalf_res.csv:1]: delete

File: careless_stats/thermolysin_dw/ccpred_overall_careless.csv
Comment (kmdalton) [inline careless_stats/thermolysin_dw/ccpred_overall_careless.csv:1]: delete

File: careless_stats/thermolysin_dw/ccpred_res.csv
Comment (kmdalton) [inline careless_stats/thermolysin_dw/ccpred_res.csv:1]: delete

File: careless_stats/thermolysin_dw/completeness.csv
Comment (kmdalton) [inline careless_stats/thermolysin_dw/completeness.csv:1]: delete

File: careless_stats/thermolysin_dw/conda_env_record.log
Comment (kmdalton) [inline careless_stats/thermolysin_dw/conda_env_record.log:1]: delete

File: careless_stats/thermolysin_dw/imagecc_res.csv
Comment (kmdalton) [inline careless_stats/thermolysin_dw/imagecc_res.csv:1]: delete

File: careless_stats/thermolysin_dw/inputs_params.log
Comment (kmdalton) [inline careless_stats/thermolysin_dw/inputs_params.log:1]: delete

File: careless_stats/thermolysin_dw/loss.png
Comment (kmdalton) [inline careless_stats/thermolysin_dw/loss.png:1]: delete

File: careless_stats/thermolysin_dw/loss_components.png
Comment (kmdalton) [inline careless_stats/thermolysin_dw/loss_components.png:1]: delete

File: careless_stats/thermolysin_dw/myoutput_19756592.err
Comment (kmdalton) [inline careless_stats/thermolysin_dw/myoutput_19756592.err:1]: delete

File: careless_stats/thermolysin_dw/myoutput_19756592.out
Comment (kmdalton) [inline careless_stats/thermolysin_dw/myoutput_19756592.out:1]: delete

File: careless_stats/thermolysin_dw/pd_completeness.png
Comment (kmdalton) [inline careless_stats/thermolysin_dw/pd_completeness.png:1]: delete

File: careless_stats/thermolysin_dw/rsplit_res.csv
Comment (kmdalton) [inline careless_stats/thermolysin_dw/rsplit_res.csv:1]: delete

File: careless_stats/thermolysin_dw/slurm_script
Comment (kmdalton) [inline careless_stats/thermolysin_dw/slurm_script:1]: delete

File: careless_stats/thermolysin_dw/thl_1p8A_grid_scale.data-00000-of-00001
Comment (kmdalton) [inline careless_stats/thermolysin_dw/thl_1p8A_grid_scale.data-00000-of-00001:1]: delete

File: careless_stats/thermolysin_dw/thl_1p8A_grid_scale.index
Comment (kmdalton) [inline careless_stats/thermolysin_dw/thl_1p8A_grid_scale.index:1]: delete

File: careless_stats/thermolysin_dw/thl_1p8A_grid_structure_factor.data-00000-of-00001
Comment (kmdalton) [inline careless_stats/thermolysin_dw/thl_1p8A_grid_structure_factor.data-00000-of-00001:1]: delete

File: careless_stats/thermolysin_dw/thl_1p8A_grid_structure_factor.index
Comment (kmdalton) [inline careless_stats/thermolysin_dw/thl_1p8A_grid_structure_factor.index:1]: delete

File: test.md
Comment (kmdalton) [inline test.md:1]: delete

**Review by kmdalton — CHANGES_REQUESTED** [review]
I think this is headed in the right direction, but we need to strip it down considerably. The purpose of this repo is education not data archiving. So, we don't need a lot of the details you might include in, for instance, a Zenodo deposition. I've done my best to mark what I think should be cut, but let me know if you have questions or if I'm missing the point. Thanks, @hkwang

**Review by kmdalton — CHANGES_REQUESTED** [review]
this is moving along please see to the following
 - double check that all the remaining files are used / needed. 
 - refactor the images and mtz files into separate subdirectories. make one directory for files users need as inputs for the various tutorials and one directory for images needed to render the markdown. 
 - remove test.md
 - add a link to this tutorial in the main README.md

**Comment by hkwang** [comment]
I: 
- double-checked the remaining files are all used. I removed one of my loss plots. I also removed `test.md` (my bad!)
- made a directory `results` for plots and text files for the `.md`, and kept `thermolysin_dw` as the directory for inputs. 
- moved the `.md` outside the example folder, to match the directory level of the other `.md` files, and created a link in the main `README.md`

**Comment by DHekstra** [comment]
can we get this PR finalized?

**Review by kmdalton — CHANGES_REQUESTED** [review]
The only change I would request is that you unlog the x-axis is loss_components.png
